src/lib/sdk/index.ts

```diff
@@ -0,0 +1,238 @@
+import qs from "qs"
+
+export type Config = {
+  baseUrl: string
+  globalHeaders?: Record<string, string>
+  publishableKey?: string
+  apiKey?: string
+  jwtToken?: {
+    storageKey?: string
+    storageMethod?: "local" | "session" | "memory"
+  }
+}
+
+type FetchParams = Parameters<typeof fetch>
+
+type ClientFetch = (
+  input: FetchParams[0],
+  init?: FetchParams[1] & { query?: Record<string, any> }
+) => Promise<Response>
+
+const isBrowser = () => typeof window !== "undefined"
+
+const toBase64 = (str: string) => {
+  if (typeof window !== "undefined") {
+    return window.btoa(str)
+  }
+
+  return Buffer.from(str).toString("base64")
+}
+
+// TODO: Add support for retries and timeouts
+class Client {
+  public fetch: ClientFetch
+
+  private DEFAULT_JWT_STORAGE_KEY = "medusa_auth_token"
+  private token = ""
+
+  constructor(config: Config) {
+    this.fetch = this.initClient(config)
+  }
+
+  protected initClient(config: Config) {
+    const defaultHeaders = {
+      "Content-Type": "application/json",
+      Accept: "application/json",
+      ...this.getApiKeyHeader(config),
+      ...this.getPublishableKeyHeader(config),
+      ...config.globalHeaders,
+    }
+
+    return (
+      input: FetchParams[0],
+      init?: FetchParams[1] & { query?: Record<string, any> }
+    ) => {
+      // We always want to fetch the up-to-date JWT token before firing off a request.
+      const jwtToken = this.getJwtTokenHeader(config)
+
+      const headers = init?.headers
+        ? { ...defaultHeaders, ...jwtToken, ...init.headers }
+        : { ...defaultHeaders, ...jwtToken }
+
+      let normalizedInput: RequestInfo | URL = input
+      if (input instanceof URL || typeof input === "string") {
+        normalizedInput = new URL(input, config.baseUrl)
+        if (init?.query) {
+          const existing = qs.parse(normalizedInput.search)
+          const stringifiedQuery = qs.stringify({ existing, ...init.query })
+          normalizedInput.search = stringifiedQuery
+        }
+      }
+
+      // TODO: Make response a bit more user friendly (throw errors, return JSON if resp content type is json, etc.)
```

Comment:
We will need to decide what the behavior of the our SDKs fetch should be when it comes to the responses, either here or when I open the PR to `medusa-js`